packages/runtime/lifecycle/src/generic/payouts.ts

```diff
@@ -38,71 +24,39 @@ import * as Rest from "@marlowe.io/runtime-rest-client";
 import { DecodingError } from "@marlowe.io/adapter/codec";
 import { stringify } from "json-bigint";
 
-class ContractLifecycle implements ContractsAPI {
-  constructor(private wallet: WalletAPI, private rest: RestAPI) {}
-  async create(req: CreateRequest): Promise<ContractId> {
-    return unsafeTaskEither(Command.create(this.rest)(this.wallet)(req));
-  }
-
-  async applyInputs(
-    contractId: ContractId,
-    provideInput: ProvideInput
-  ): Promise<ContractId> {
-    const contractDetails = await unsafeTaskEither(
-      this.rest.contracts.contract.get(contractId)
-    );
-    const parties = await getParties(this.wallet)(
-      contractDetails.roleTokenMintingPolicyId
-    )();
-    const next = await unsafeTaskEither(
-      this.rest.contracts.contract.next(contractId)(
-        mkEnvironment(pipe(Date.now(), (date) => subMinutes(date, 15)))(
-          pipe(Date.now(), (date) => addMinutes(date, 15))
-        )
-      )(parties)
-    );
-
-    return unsafeTaskEither(
-      Command.applyInputs(this.rest)(this.wallet)(contractId)(
-        provideInput(next)
-      )
-    );
-  }
+export function mkPayoutLifecycle(
```

Comment:
Didn't noticed this before, but this should also receive (and pass) the DI, no need to manually pass each component and construct it here,.